// @parthi033/ui-foundation/dist/collection/components/pn-button/pn-button.js
import { h } from "@stencil/core";
export class PnButton {
    label;
    type = 'primary';
    rounded = false;
    shadow = false;
    iconLeft;
    iconRight;
    iconOnly = false;
    render() {
        const isDisabled = this.type === 'disabled';
        return (h("button", { key: '3251614e1f480c01d4dc0479638c721e96934834', class: `btn 
                    ${this.type} 
                    ${this.rounded ? 'rounded' : ''}
                    ${this.shadow ? 'shadow' : ''}
                    ${this.iconLeft ? 'has-icon-left' : ''}
                    ${this.iconRight ? 'has-icon-right' : ''}
                    ${this.iconOnly ? 'icon-only' : ''}
                `, disabled: isDisabled }, this.iconLeft && (h("i", { key: 'bf4ac064e36c9a7d27c50989f9c18f3f7b46d15a', class: `btn-icon btn-icon-left ${this.iconLeft}` })), !this.iconOnly && this.label && (h("span", { key: '689034b7d2857dea03e6d722117a533e861370a3', class: "btn-text" }, this.label)), this.iconRight && (h("i", { key: '28521aa32c1c338c5c424b34e4a1bd8737e280f4', class: `btn-icon btn-icon-right ${this.iconRight}` }))));
    }
    static get is() { return "pn-button"; }
    static get originalStyleUrls() {
        return {
            "$": ["pn-button.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["pn-button.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "label"
            },
            "type": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'secondary' | 'tertiary' | 'white' | 'disabled'",
                    "resolved": "\"disabled\" | \"primary\" | \"secondary\" | \"tertiary\" | \"white\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "type",
                "defaultValue": "'primary'"
            },
            "rounded": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "rounded",
                "defaultValue": "false"
            },
            "shadow": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "shadow",
                "defaultValue": "false"
            },
            "iconLeft": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "icon-left"
            },
            "iconRight": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "icon-right"
            },
            "iconOnly": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "icon-only",
                "defaultValue": "false"
            }
        };
    }
}
