// @parthi033/ui-foundation/dist/collection/components/pn-button/pn-button.js
import { h } from "@stencil/core";
export class PnButton {
    label;
    type = 'primary';
    rounded = false;
    render() {
        const isDisabled = this.type === 'disabled';
        return h("button", { key: 'cbe817a5769ef625f4bc46b9b6224884c40e4d17', class: `btn ${this.type} ${this.rounded ? 'rounded' : ''}`, disabled: isDisabled }, this.label);
    }
    static get is() { return "pn-button"; }
    static get originalStyleUrls() {
        return {
            "$": ["pn-button.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["pn-button.css"]
        };
    }
    static get properties() {
        return {
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "label"
            },
            "type": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "'primary' | 'secondary' | 'disabled'",
                    "resolved": "\"disabled\" | \"primary\" | \"secondary\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "type",
                "defaultValue": "'primary'"
            },
            "rounded": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "reflect": false,
                "attribute": "rounded",
                "defaultValue": "false"
            }
        };
    }
}
